parsec/core/gui/app.py

```diff
@@ -153,9 +181,11 @@ def kill_window(*args):
         # QTimer wakes up the event loop periodically which allows us to close
         # the window even when it is in background.
         timer = QTimer()
-        timer.start(400)
-        timer.timeout.connect(lambda: None)
+        timer.start(1000 if diagnose else 400)
```

Comment:
Not sure about this... In diagnostic mode we run parsec for 1s, then stop concluding everything was fine.
In theory 1s is far enough for the application to start, but considering what antivirus like Avast are doing I would feel better if we start waiting after the application has reached a certain idle state

Reply:
Good point! The tricky thing with this command is that we want to test the most out of it while interacting as little as possible with the regular workflow (in order to avoid false positive). It's far from perfect at the moment and it can be improved over time.

Still I'll have a quick look at it and see if I can find a better stop condition for the diagnose run.

Reply:
👍

Reply:
From the Qt docs on [app.exec](https://doc.qt.io/qt-5/qcoreapplication.html#exec):
> To make your application perform idle processing (by executing a special function whenever there are no pending events), use a QTimer with 0 timeout. More advanced idle processing schemes can be achieved using processEvents().

A timer seems to be ok for this purpose. I'll leave it as it is for the moment.